x-pack/plugins/security_solution/public/flyout/document_details/right/components/graph_preview.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiSkeletonText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { Graph } from '@kbn/cloud-security-posture-graph';
+import type {
+  NodeDataModel,
+  EdgeDataModel,
+} from '@kbn/cloud-security-posture-common/types/graph/latest';
+import { GRAPH_PREVIEW_TEST_ID, GRAPH_PREVIEW_LOADING_TEST_ID } from './test_ids';
+
+export interface GraphPreviewProps {
+  isLoading: boolean;
+  isError: boolean;
+  data?: {
+    nodes: NodeDataModel[];
+    edges: EdgeDataModel[];
+  };
+}
+
+/**
+ * Graph preview under Overview, Visualizations. It shows a graph without abilities to expand.
+ */
+export const GraphPreview: React.FC<GraphPreviewProps> = ({
```

Comment:
do we want to memoize this component?

Reply:
we can, I wasn't sure about the effects of it yet (kinda new to react).

I can use `memo` and see if works